data_structures/array_list.py

```diff
@@ -86,7 +86,7 @@ def __shuffle_left(self, index: int) -> None:
         :complexity worst: O(N) shuffle from the start of the list
         where N is the number of items in the list
         """
-        for i in range(index, len(self)):
+        for i in range(index, len(self) - 1):
```

Comment:
Ah this and moving the length decrement above are simple changes that make it much clearer for students and easier to explain too. Thanks!